src/ucp/rndv/rndv.c

```diff
@@ -841,10 +849,18 @@ ucp_rndv_send_frag_get_mem_type(ucp_request_t *sreq, ucs_ptr_map_key_t rreq_id,
         return UCS_ERR_NO_MEMORY;
     }
 
-    mdesc = ucp_worker_mpool_get(&worker->rndv_frag_mp);
-    if (ucs_unlikely(mdesc == NULL)) {
-        ucs_error("failed to allocate fragment memory desc");
-        return UCS_ERR_NO_MEMORY;
+    /* get mpool for mem_type */
+    mpool_key.mem_type  = UCS_MEMORY_TYPE_HOST;
+    mpool_key.device_id = -1;
+
+    frag_mp = ucp_worker_get_mem_type_mpool(worker, &mpool_key);
+    if (frag_mp == NULL) {
+        ucs_fatal("no frag memory pool");
+    }
+
+    mdesc = ucp_worker_mpool_get(frag_mp);
+    if (mdesc == NULL) {
+        ucs_fatal("failed to allocate fragment memory buffer");
```

Comment:
define a helper func "ucp_worker_mem_type_mpool_get(worker, mem_type)" and return the desc.
and reuse it everywhere

Reply:
"mdesc->memh         = mem_type_mdesc->memh"
this can be inside the helper func

Reply:
Is this safe usage? At some point the original memh of mdesc needs to be unregistered right? By overwriting we probably don't do that with the current changes.

Reply:
this should happen when you free(unmap) the frag pool.

Reply:
can we have a helper function?